### PR TITLE
GDB-8105: YASR: Error handling

### PR DESCRIPTION
**Pull request by boyan-tonchev**
## What
Added an error plugin that displays error messages when something goes wrong during query execution.

## Why
WB has such functionality that displays a message that describes to the user what the reason for which the exception occurred.

## How
A new "extended_error" plugin is added. It has responsibility to displays the error messages that can be occurred during query execution.
Added error info to yasr persistence.

Additional work:
-Rename the registration name of the ExtendedResponse plugin to be similar like other extended plugins.
-Rename "$background-onto-default" to "$background-color-onto-default".
- Remove a test form language suite because, after error plugin implementation, it doesn't sense.

### CODE REVIEW

File: Yasgui/packages/yasr/src/plugins/error/extended-error.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    errorResponseElement.innerHTML = this.getErrorMessage(error);
+    return errorResponseElement;
+  }
+
+  private getErrorMessage(error: Parser.ErrorSummary): string {
+    return (
+      `<div class="error-response-plugin-header">` +
+      `<div class="error-response-plugin-error-status">${error.status ? error.status + ":" : ""} ${
+        error.statusText
+      }</div>` +
```
Comment (svilenvelikov) [inline Yasgui/packages/yasr/src/plugins/error/extended-error.ts:24]: When you use a string literal, you don't need to do a string concatenation with `+`

File: Yasgui/packages/yasgui/src/Tab.ts [hunk 2]
```diff
@@ -506,13 +506,14 @@ export class Tab extends EventEmitter {
...
+    const responseAsStoreObject = this.yasr.results.getAsStoreObject(this.yasgui.config.yasr.maxPersistentResponseSize);
     if (!this.yasr.results.hasError()) {
-      this.persistentJson.yasr.response = this.yasr.results.getAsStoreObject(
-        this.yasgui.config.yasr.maxPersistentResponseSize
-      );
+      this.persistentJson.yasr.response = responseAsStoreObject;
     } else {
-      // Don't persist if there is an error and remove the previous result
-      this.persistentJson.yasr.response = undefined;
+      if (responseAsStoreObject && responseAsStoreObject.error && responseAsStoreObject.error.text) {
```
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply Yasgui/packages/yasgui/src/Tab.ts:513]: It cannot actually be undefined. According to the `ErrorSummary` interface, this is a required field. I will remove this check.

File: Yasgui/packages/yasr/src/extended-yasr.ts [hunk 3]
```diff
@@ -63,14 +63,16 @@ export class ExtendedYasr extends Yasr {
     }
   }
 
-  updatePluginSelectorNames() {
-    if (this.yasqe.isAskQuery()) {
-      // hides plugins buttons if "query" mode query had been executed.
+  draw() {
+    if (this.yasqe.isUpdateQuery() || this.yasqe.isAskQuery() || this.results?.hasError()) {
```
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply Yasgui/packages/yasr/src/extended-yasr.ts:67]: When results is undefined, the "hidden" class should be removed from plugins buttons. And then the execution of the parent `super.draw()` function will take care of what to do.

File: Yasgui/packages/yasr/src/extended-yasr.ts [hunk 4]
```diff
@@ -305,6 +307,13 @@ export class ExtendedYasr extends Yasr {
 
   updateResponseInfo() {
     const responseInfoElement = this.getResponseInfoElement();
+
+    removeClass(responseInfoElement, "hidden");
+    if (this.results?.hasError()) {
```
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply Yasgui/packages/yasr/src/extended-yasr.ts:312]: if there are no results, then there is no error, so we don't need to add a hidden class.

File: Yasgui/packages/yasr/src/extended-yasr.ts [hunk 5]
```diff
@@ -305,6 +307,13 @@ export class ExtendedYasr extends Yasr {
...
+      addClass(responseInfoElement, "hidden");
+      return;
```
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply Yasgui/packages/yasr/src/extended-yasr.ts:314]: If the "hidden" class is added to the `responseInfoElement` element, it doesn't need to execute the rest of the function.

[the remark this answers is not in the document]
Comment (svilenvelikov) [reply Yasgui/packages/yasr/src/extended-yasr.ts:314]: but there isn't anything else in the function :)

[the remark this answers is not in the document]
Comment (svilenvelikov) [reply Yasgui/packages/yasr/src/extended-yasr.ts:314]: Sorry. My bad. Didn't paid attention that this is not the entire function.

**Review by svilenvelikov — APPROVED** [review]

**Pull request merged**